searx/engines/wordnik.py

```diff
@@ -0,0 +1,126 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""
+ Wordnik (general)
+"""
+
+import re
+from lxml.html import fromstring
+from searx.utils import extract_text
+from searx.raise_for_httperror import raise_for_httperror
+
+# about
+about = {
+    "website": 'https://www.wordnik.com',
+    "wikidata_id": 'Q8034401',
+    "official_api_documentation": None,
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": 'HTML',
+}
+
+categories = ['general']
+paging = False
+
+word_xpath = '//*[@id="headword"]/text()'
+from_xpath = '//*[@id="define"]/div/h3[1]'
+definitions_xpath = '//*[@id="define"]/div/ul[1]'
+misspelling_xpath = '//*[@id="define"]/div/ul/li/xref'
+
+URL = 'https://www.wordnik.com'
+SEARCH_URL = URL + '/words/{query}'
+
+
+def request(query, params):
+    params['url'] = SEARCH_URL.format(query=query)
+    return params
+
+
+def response(resp):
+    try:
+        dom = fromstring(resp.text)
+    except Exception:
+        raise_for_httperror(resp)
```

Comment:
This is only for HTTP errors (see my branch)

Reply:
Lots to be learnt. Will get there eventually.